src/base/ini_parser.c

```diff
@@ -517,7 +517,11 @@ ini_parse_line (char *input_line, char *section, char *key, char *value)
 	{
 	  sprintf (section, "%c%s", leading_char, ini_str_trim (section + 1));
 	}
-      strcpy (section, ini_str_lower (section));
+
```

Comment:
You should also check that leading_char is not %.

Reply:
I think hostname itself is case insensitive, localhost or LOCALhost are the same host.
Should I consider '%'?

Reply:
You are right. so you don't need to consider it.